parse.go

```diff
@@ -222,21 +225,32 @@ type ParseOption func(*parseOptSet)
 
 // parseOptSet represents the flattened option set after all ParseOptions have been applied.
 type parseOptSet struct {
-	skipMetadataReadOnNewParserInit    bool
-	allowMismatchPixelDataLength       bool
-	skipPixelData                      bool
-	skipProcessingPixelDataValue       bool
-	allowMissingMetaElementGroupLength bool
+	skipMetadataReadOnNewParserInit                bool
+	allowMismatchPixelDataLength                   bool
+	skipPixelData                                  bool
+	skipProcessingPixelDataValue                   bool
+	allowMissingMetaElementGroupLength             bool
+	tagSpecificCharacterSetToEncodingNameConverter func(string) string
```

Comment:
To archive maximum plexibility, it makes more sense if the flag output is []*encoding.Decoder

```go
type parseOptSet struct {
   getEncoderFn func([]string) ([]*encoding.Decoder, error)
}

func ParseSpecificCharacterSet(encodingNames []string, fn ...) {
  var decoders []*encoding.Decoder
  var error
  if fn == nil {
     decoders, err = DefaultGetEncoder(encodingNames)
  } else {
    decoders, err = fn(encodingNames)
  }
...
}


```

Reply:
updated at https://github.com/segmed/dicom/pull/2/commits/d90c4f7ebe371515c99a7441551d53f4d3b4d6ec  https://github.com/segmed/dicom/pull/2/commits/f3bf6af00a697206b06b61ce7231543dd9013708